codeReview

ashraffares opened the pull request:
milestone1
milestone2
milestone3
milestone4
milestone5

ashraffares committed:
Update readme

ashraffares commented:
I have been told via email from Willow to submit all milestones for the bookstore, and all of the milestones are gonna be reviewed at once because my dashboard was broken and I was not able to submit them.
thanks a lot, @Aziz for giving me this exception.

addod19 commented on README.md:
```diff
@@ -1,5 +1,5 @@
 ## BookStore App
-"BookStore App"
+"BookStore App" is the front-end of an application that organizes and manage books in bookstore.
```
- [ ] It would be better if we improve our PR. It should be professional, PR is a way of communicating the work we did. Instead of writing `milestone1............milestone2`, we could make it a little professional, talk about the main features, styling, etc. We can find information from the project description

addod19 commented on README.md:
```diff
@@ -1,5 +1,5 @@
 ## BookStore App
-"BookStore App"
```
- [ ] It would be better to achieve the desired styling for the project. We are missing the user icon, also the Bookstore title and the user icon must align with the Bookform component

addod19 commented on README.md:
```diff
@@ -1,5 +1,5 @@
 ## BookStore App
```
### Optional
- [ ] It would be better to capitalize our commit messages, this makes our project professional. Also, let us commit more often

rukundoeric reviewed (approved):
Hi @ashraffares,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------

addod19 commented on README.md:
```diff
@@ -1,5 +1,5 @@
 ## BookStore App
-"BookStore App"
+"BookStore App" is the front-end of an application that organizes and manage books in bookstore.
 
```
- [ ] It would be better if we implement the DRY principle. `display: flex` has been repeated in the `navbar.css` file. Also `border-radius: 4px;`, `padding: 15px;`, `border: solid 1px #e8e8e8;`, `color: #c4c4c4;`, has been repeated int the `bookform.css` file. There are a couple of repeating rules in the `book.css` file

ashraffares replied:
You do understand that this project has already been reviewed by a code reviewer and even though why did you request changes on milestone2 and it got nothing to do with styling

ashraffares replied:
every component has its own CSS I'm not gonna use a different CSS file for another component this gonna make it harder to debug  and  harder to understand where is styling coming from 
and this is the recommended way for adding CSS module and here's the docs
https://create-react-app.dev/docs/adding-a-css-modules-stylesheet

addod19 reviewed (changes requested):
### Hi @ashraffares, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

__Happy coding__ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

addod19 commented:
> I have been told via email from Willow to submit all milestones for the bookstore, and all of the milestones are gonna be reviewed at once because my dashboard was broken and I was not able to submit them.
> thanks a lot, @aziz for giving me this exception.

Noted....This project is reviewed based on the comment

rukundoeric reviewed (approved):
Hi @ashraffares,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As mentioned by the student in the above comment, all milestones are submitted at once in this PR. **But my review is only about milestone 4**

rukundoeric commented:
Hi @ashraffares,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As mentioned by the student in the above comment, all milestones are submitted at once in this PR. **But my review is only about milestone 2**

taiwo2 reviewed (changes requested):
# STATUS: CHANGES REQUIRED ♻️ 
@ashraffares well done on this project I commend your effort, I reviewed this project based on milestone 5, kindly implement the suggested changes by the previous Code Reviewer, however, changes are required, and also the project stated thus

- Add all element in the design, features that haven’t been implemented yet should be added as a placeholder and does not need to be implemented
- Styling is EXACTLY the same as the one provided in the designs

`your app`
![Screenshot 2021-09-09 174344](https://user-images.githubusercontent.com/38242590/132728744-9df793f3-caf0-4e64-8d6a-e1aa0ac1bfef.png)
`project design`
![Screenshot 2021-09-06 214133](https://user-images.githubusercontent.com/38242590/132730167-1dc12d9d-f3a6-4301-bbe8-62098575635d.png)


- [ ] It would be better to achieve the desired styling for the project. We are missing the user icon
- [ ] It would be better if we implement the DRY principle. display: flex has been repeated in the navbar.css,book.css file. Also `border-radius: 4px;, padding: 15px;, border: solid 1px #e8e8e8;, color: #c4c4c4;`, has been repeated in the bookform.css file. There are a couple of repeating rules in the book.css file and all other related repeated styles. kindLy fix
- [ ]   I can see `comment` and `edit` in your app kindly fix

### once specific changes are made kindly request for review

ashraffares committed:
Add use Icon

ashraffares committed:
Add comments and edit

ashraffares commented:
Added comment - edit buttons.
Added user icon.
Added placeholder Author.

![React App (1)](https://user-images.githubusercontent.com/37639594/132739251-14335f92-f739-453e-9935-b48d1b4de98f.png)

rukundoeric reviewed (changes requested):
Hi @ashraffares, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  - [ ] As mentioned by the previous reviewer, your app still doesn't look like the mockup design.
  
| Original      | Yours |
  | ----------- | ----------- |
| ![80047697-0c3f6280-8506-11ea-9416-a4f1800e2329](https://user-images.githubusercontent.com/45217364/130871998-88863e2f-c823-4b7b-a5c4-20162973ff21.png) | ![scrnli_9_9_2021_8-10-10 PM](https://user-images.githubusercontent.com/45217364/132740388-6d6a75c0-73cc-452a-90ec-8fc4f7034b23.png) |
- Category filter should be in the toolbar as it is on the mockup design.
- In your design, there are some margins and spacing that are not present in the mockup. kindly fix them.
- There should be a horizontal line between the book list and the form.
- Your interface is missing a profile avatar icon and a title for the form. kindly add them.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - N/A 

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

ashraffares commented:
Please understand this project has already been reviewed and approved by the code reviewer, you must understand how much stressful I'm to do a code review for a project I already did.
my project doesn't have any logical errors and I'm following the design guidelines. 

this is the link of my partner that I collaborated with on this project we have the same code the same design and the project is approved https://github.com/vidhishweta01/bookstores

rukundoeric commented:
> Please understand this project has already been reviewed and approved by the code reviewer, you must understand how much stressful I'm to do a code review for a project I already did.
> my project doesn't have any logical errors and I'm following the design guidelines.
> 
> this is the link of my partner that I collaborated with on this project we have the same code the same design and the project is approved https://github.com/vidhishweta01/bookstores

Yeah, I understand that. but I'm following guidelines too. 
- Styling should be **EXACTLY** the same as the one provided in the designs ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone5/exactlySameAsDesign.png) from [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/45791599/))

That is one of the points I must check when reviewing this project milestone 5. It's okay, since you invested a lot of time, I will go ahead and mark this project as `approved`. but remember that when a client gives you a project, you are expected to deliver the project exactly as they requested it with no excuses.

rukundoeric commented:
Hi @ashraffares,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
